cordapp/src/main/kotlin/net/corda/plugins/SigningOptions.kt

```diff
@@ -2,207 +2,160 @@ package net.corda.plugins
+            val KEYPASS = "keypass"
+            val SIGFILE = "sigfile"
+            val SIGNEDJAR = "signedjar"
+            val VERBOSE = "verbose"
+            val STRICT = "strict"
+            val INTERNALSF = "internalsf"
+            val SECTIONSONLY = "sectionsonly"
+            val LAZY = "lazy"
+            val MAXMEMORY = "maxmemory"
+            val PRESERVELASTMODIFIED = "preservelastmodified"
+            val TSACERT = "tsaurl"
+            val TSAURL = "tsacert"
+            val TSAPROXYHOST = "tsaproxyhost"
+            val TSAPROXYPORT = "tsaproxyport"
+            val EXECUTABLE = "executable"
+            val FORCE = "force"
+            val SIGALG = "sigalg"
+            val DIGESTALG = "digestalg"
+            val TSADIGESTALG = "tsadigestalg"
+        }
     }
 
     @get:Input
-    var storepass = ""
-
-    fun storepass(value: String?) {
-        storepass = value ?: ""
-    }
+    var alias = System.getProperty(SYSTEM_PROPERTY_PREFIX + Key.ALIAS, DEFAULT_ALIAS)
 
     @get:Input
-    var keystore = ""
-
-    fun keystore(value: String?) {
-        keystore = value ?: ""
-    }
+    var storepass = System.getProperty(SYSTEM_PROPERTY_PREFIX + Key.STOREPASS, DEFAULT_STOREPASS)
 
     @get:Input
-    var storetype = ""
-
-    fun storetype(value: String?) {
-        storetype = value ?: ""
-    }
+    var keystore = System.getProperty(SYSTEM_PROPERTY_PREFIX + Key.KEYSTORE, DEFAULT_KEYSTORE)
 
     @get:Input
-    var keypass = ""
+    var storetype = System.getProperty(SYSTEM_PROPERTY_PREFIX + Key.STORETYPE, DEFAULT_STORETYPE)
 
-    fun keypass(value: String?) {
-        keypass = value ?: ""
-    }
+    @get:Input
+    var keypass = System.getProperty(SYSTEM_PROPERTY_PREFIX + Key.KEYPASS, DEFAULT_KEYPASS)
 
     @get:Input
     var sigfile = ""
 
-    fun sigfile(value: String?) {
-        sigfile = value ?: ""
-    }
-
     @get:Input
     var signedjar = ""
 
-    fun signedjar(value: String?) {
-        signedjar = value ?: ""
-    }
-
     @get:Input
     var verbose = ""
 
-    fun verbose(value: String?) {
-        verbose = value ?: ""
+    fun verbose(value: Boolean) {
+        verbose = value.toString()
     }
 
     @get:Input
     var strict = ""
 
-    fun strict(value: String?) {
-        strict = value ?: ""
+    fun strict(value: Boolean) {
+        strict = value.toString()
     }
 
     @get:Input
     var internalsf = ""
 
-    fun internalsf(value: String?) {
-        internalsf = value ?: ""
+    fun internalsf(value: Boolean) {
+        internalsf = value.toString()
     }
 
     @get:Input
     var sectionsonly = ""
 
-    fun sectionsonly(value: String?) {
-        sectionsonly = value ?: ""
+    fun sectionsonly(value: Boolean) {
+        sectionsonly = value.toString()
     }
 
     @get:Input
     var lazy = ""
 
-    fun lazy(value: String?) {
-        lazy = value ?: ""
+    fun lazy(value: Boolean) {
+        lazy = value.toString()
     }
 
     @get:Input
     var maxmemory = ""
```

Comment:
Would changing this setting affect the hash of the output artifact? If it wouldn't then the property is probably `@Internal`. (Not sure _whose_ memory is having its maximum set here, to be fair.)

Reply:
yes, it's memory to run `jarsigner`